File: spec/formatters/numbers/helpers/fraction_spec.rb
# encoding: UTF-8

# Copyright 2012 Twitter, Inc
# http://www.apache.org/licenses/LICENSE-2.0

require 'spec_helper'

describe TwitterCldr::Formatters::Numbers::Fraction do
  describe "#apply" do
    it "test: formats a fraction" do
      token = TwitterCldr::Tokenizers::Token.new(value: '###.##', type: :pattern)
      expect(described_class.new(token).apply('45')).to eq('.45')
    end

    it "test: pads zero digits on the right side" do
      token = TwitterCldr::Tokenizers::Token.new(value: '###.0000#', type: :pattern)
      expect(described_class.new(token).apply('45')).to eq('.45000')
    end

    it "test: :precision option overrides format precision" do
      token = TwitterCldr::Tokenizers::Token.new(value: '###.##', type: :pattern)
      expect(described_class.new(token).apply('78901', precision: 5)).to eq('.78901')
    end
  end
end

File: spec/js/numbers/helpers/fraction.spec.js
// Copyright 2012 Twitter, Inc
// http://www.apache.org/licenses/LICENSE-2.0

var TwitterCldr = require('../../../../lib/assets/javascripts/twitter_cldr/en.js');

describe("FractionHelper", function() {
  describe("#apply", function() {
    it("test: formats a fraction", function() {
      var token = '###.##';
      expect(new TwitterCldr.NumberFormatter.FractionHelper(token).apply('45')).toEqual('.45');
    });

    it("test: pads zero digits on the right side", function() {
      var token = '###.0000#';
      expect(new TwitterCldr.NumberFormatter.FractionHelper(token).apply('45')).toEqual('.45000');
    });

    it("test: :precision option overrides format precision", function() {
      var token = '###.##';
      expect(new TwitterCldr.NumberFormatter.FractionHelper(token).apply('78901', {precision: 5})).toEqual('.78901');
    });
  });
});